contracts/connectors/loantoken/LoanTokenLogicStandard.sol

```diff
@@ -955,6 +751,22 @@ contract LoanTokenLogicStandard is AdvancedToken, StorageExtension {
         );
     }
 
+    function _checkPermit(address token, bytes memory loanDataBytes) internal returns (bytes memory) {
+        if (loanDataBytes.length != 0) {
+            if(abi.decode(loanDataBytes, (uint128)) & WITH_PERMIT != 0) {
+                (uint128 f, bytes[] memory payload) = abi.decode(
+                    loanDataBytes,
+                    (uint128, bytes[])
+                );
+                (address owner, address spender, uint value, uint deadline, uint8 v, bytes32 r, bytes32 s) = abi.decode(payload[2], (address, address, uint, uint, uint8, bytes32, bytes32));
+                IERC20Permit(token).permit(owner, spender, value, deadline, v, r, s);
+                payload[0] = "";
+                loanDataBytes = abi.encode(f, payload);
+            }
+            return loanDataBytes;
```

Comment:
move return statement to outside both if statements. There is a possibility it could wipe loanDataBytes passed if permit is not selected as it is called in all cases for borrow and marginTrade

Reply:
fixed